core/chains/evm/txmgr/eth_confirmer_test.go

```diff
@@ -1256,79 +1255,20 @@ func TestEthConfirmer_CheckConfirmedMissingReceipt_smallEvmRPCBatchSize_middleBa
 }
 
-func TestEthConfirmer_FindEthTxsRequiringResubmissionDueToInsufficientEth(t *testing.T) {
-	t.Parallel()
-
-	db := pgtest.NewSqlxDB(t)
-	cfg := configtest.NewGeneralConfig(t, nil)
-	borm := cltest.NewTxmORM(t, db, cfg)
-	q := pg.NewQ(db, logger.TestLogger(t), cfg)
-
-	ethKeyStore := cltest.NewKeyStore(t, db, cfg).Eth()
-
-	_, fromAddress := cltest.MustInsertRandomKeyReturningState(t, ethKeyStore, 0)
-	_, otherAddress := cltest.MustInsertRandomKeyReturningState(t, ethKeyStore, 0)
-
-	// Insert order is mixed up to test sorting
-	etx2 := cltest.MustInsertUnconfirmedEthTxWithInsufficientEthAttempt(t, borm, 1, fromAddress)
-	etx3 := cltest.MustInsertUnconfirmedEthTxWithBroadcastLegacyAttempt(t, borm, 2, fromAddress)
-	attempt3_2 := cltest.NewLegacyEthTxAttempt(t, etx3.ID)
-	attempt3_2.State = txmgr.EthTxAttemptInsufficientEth
-	attempt3_2.GasPrice = assets.NewWeiI(100)
-	require.NoError(t, borm.InsertEthTxAttempt(&attempt3_2))
-	etx1 := cltest.MustInsertUnconfirmedEthTxWithInsufficientEthAttempt(t, borm, 0, fromAddress)
-
-	// These should never be returned
-	cltest.MustInsertUnconfirmedEthTxWithBroadcastLegacyAttempt(t, borm, 3, fromAddress)
-	cltest.MustInsertConfirmedEthTxWithLegacyAttempt(t, borm, 4, 100, fromAddress)
-	cltest.MustInsertUnconfirmedEthTxWithInsufficientEthAttempt(t, borm, 0, otherAddress)
-
-	t.Run("returns all eth_txes with at least one attempt that is in insufficient_eth state", func(t *testing.T) {
-		etxs, err := txmgr.FindEthTxsRequiringResubmissionDueToInsufficientEth(testutils.Context(t), q, logger.TestLogger(t), fromAddress, cltest.FixtureChainID)
-		require.NoError(t, err)
-
-		assert.Len(t, etxs, 3)
-
-		assert.Equal(t, *etx1.Nonce, *etxs[0].Nonce)
-		assert.Equal(t, etx1.ID, etxs[0].ID)
-		assert.Equal(t, *etx2.Nonce, *etxs[1].Nonce)
-		assert.Equal(t, etx2.ID, etxs[1].ID)
-		assert.Equal(t, *etx3.Nonce, *etxs[2].Nonce)
-		assert.Equal(t, etx3.ID, etxs[2].ID)
-	})
-
-	t.Run("does not return eth_txes with different chain ID", func(t *testing.T) {
-		etxs, err := txmgr.FindEthTxsRequiringResubmissionDueToInsufficientEth(testutils.Context(t), q, logger.TestLogger(t), fromAddress, *big.NewInt(42))
-		require.NoError(t, err)
-
-		assert.Len(t, etxs, 0)
-	})
-
-	t.Run("does not return confirmed or fatally errored eth_txes", func(t *testing.T) {
-		pgtest.MustExec(t, db, `UPDATE eth_txes SET state='confirmed' WHERE id = $1`, etx1.ID)
-		pgtest.MustExec(t, db, `UPDATE eth_txes SET state='fatal_error', nonce=NULL, error='foo', broadcast_at=NULL, initial_broadcast_at=NULL WHERE id = $1`, etx2.ID)
-
-		etxs, err := txmgr.FindEthTxsRequiringResubmissionDueToInsufficientEth(testutils.Context(t), q, logger.TestLogger(t), fromAddress, cltest.FixtureChainID)
-		require.NoError(t, err)
-
-		assert.Len(t, etxs, 1)
-
-		assert.Equal(t, *etx3.Nonce, *etxs[0].Nonce)
-		assert.Equal(t, etx3.ID, etxs[0].ID)
-	})
-}
-
 func TestEthConfirmer_FindEthTxsRequiringRebroadcast(t *testing.T) {
```

Comment:
I would generally prefer to have all tests which inspect the database directly to ensure correctness, to be inside orm_test.go.

But since we have this structure of many tests in this file actually testing the orm behavior as legacy, i guess its ok to leave them as it is.

Reply:
I think it's ok to use the orm in other test files for generating data. But yeah, if we want to decouple database inspection for test verification then we'd have to drastically change our testing approach in eth confirmer to not assert on database results at all. We'd mainly be testing return results and mocking database calls to test pure service logic

Reply:
Yes, that would be the right way. So we will only validate that the confirmer is calling the ORM correctly.
But ok for now.